### PR TITLE
Update samples for lifecycle 0.4.0 (API 0.2)

### LINKED ISSUE #30: Update samples to work with pack/lifecycle v0.4.0
[PLACEHOLDER]

### PR DESCRIPTION
**Pull request by jromero**
* Update buildpacks to API v0.2
* Add [java app](https://github.com/buildpack/sample-java-app) built with maven sample
* Add sample [stack](https://github.com/buildpack/stacks)

Resolves #30

### CODE REVIEW

File: apps/java-maven/pom.xml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>org.springframework.boot</groupId>
+		<artifactId>spring-boot-starter-parent</artifactId>
+		<version>2.1.3.RELEASE</version>
```
Comment (jkutner) [inline apps/java-maven/pom.xml]: can we use a version range here to ensure that we're not falling behind on important security patches to the sample? (I'm not sure if version ranges like `[2.1.3,)` work with parent poms)

**Comment by jkutner** [comment]
I wonder how necessary the sample app is. I imagine most people will want to run against their own app, but I suppose the sample buildpack will fail more often than the hardened CF and Heroku Java buildpacks.

**Comment by jromero** [comment]
> I wonder how necessary the sample app is. I imagine most people will want to run against their own app, but I suppose the sample buildpack will fail more often than the hardened CF and Heroku Java buildpacks.

I thought about the same but I found that having the sample apps puts a nice bow on the whole repo and tutorials. Because these sample apps exist we can do things like having more complete executable commands in README files and good references for tutorials (see Ruby tutorial).

**TLDR;** In my mind the benefits below lower the bar of entry.

**Benefits:**

1. We have [tutorials](https://buildpacks.io/docs/app-journey/) that use these apps. _(we'll be add the Ruby sample app as well)_
2. Allows for [executable examples](https://github.com/buildpack/samples/tree/feature/30-update-samples-lifecycle-0.4.0/buildpacks/java-maven)
3. We should refrain from having them execute anything against _their_ apps because of the reasons you mentioned. _(these aren't comprehensive and robust buildpacks)_
4. Conformance tests (#31) would utilize sample apps.

File: apps/java-maven/pom.xml (hunk 1 above)
Comment (jromero) [reply apps/java-maven/pom.xml]: That's a really good idea. I'll look into it.

Comment (jromero) [reply apps/java-maven/pom.xml]: Worked like a charm. Updated!
